internal/semver/semver.go

```diff
@@ -0,0 +1,53 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package semver provides functions related to semanic version.
+// This package is based on "golang.org/x/mod/semver"
+package semver
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/mod/semver"
+)
+
+// IsSemverValid returns true if version is a valid semantic version
+func IsSemverValid(version string) bool {
+	// a valid semanic version MUST not have prefix 'v'
+	if strings.HasPrefix(version, "v") {
+		return false
+	}
+	// golang package "golang.org/x/mod/semver" requires prefix 'v'
+	return semver.IsValid("v" + version)
```

Comment:
Please use the old regex method to check valid semver. 
Golang's `semver.IsValid`has  two exceptions. First, it requires the "v" prefix, which we addressed. Second, it recognizes vMAJOR and vMAJOR.MINOR (with no prerelease or build suffixes) as shorthands for vMAJOR.0.0 and vMAJOR.MINOR.0 and this one isnt addressed.

Reply:
> this one isnt addressed.

@priteshbandi which one isn't addressed by `semver.IsValid` again? (when I run `semver.IsValid("v1.2.0")` or `semver.IsValid("v1.2.3")`, they return true.)

Reply:
Try for `semver.IsValid("v1")` or `semver.IsValid("v1.2")`.

Reply:
@priteshbandi I see. From https://semver.org/#backusnaur-form-grammar-for-valid-semver-versions, there's:
```
<valid semver> ::= <version core>
                 | <version core> "-" <pre-release>
                 | <version core> "+" <build>
                 | <version core> "-" <pre-release> "+" <build>

<version core> ::= <major> "." <minor> "." <patch>
```
 Changing back to the old regex check. /cc: @shizhMSFT